pkg/queue/sharedmain/main.go

```diff
@@ -303,6 +314,11 @@ func Main(opts ...Option) error {
 				logger.Errorw("Failed to shutdown server", zap.String("server", name), zap.Error(err))
 			}
 		}
+
+		if certWatcher != nil {
+			certWatcher.Stop()
+		}
```

Comment:
make this a `defer` call after we successfully start it - or pass it a context that will stop the cert watcher when it's cancelled